### PR TITLE
Always transplant a stub for transplanted methods

### LINKED ISSUE #1845: docs for many AtomGroup methods missing
### Expected behaviour
radius_of_gyration(), principal_axes() (see #1828 ), center_of_mass() and most other "compute something on an AtomGroup" (that require e.g., mass) should be in the docs for AtomGroup.

### Actual behaviour
They are not in the docs anymore (at least since 0.17.0 but probably since #363 / 0.16.0)

### Code to reproduce the behaviour
Look at https://www.mdanalysis.org/mdanalysis/documentation_pages/core/groups.html#MDAnalysis.core.groups.AtomGroup

I think the problem is that we only attached certain methods to AtomGroup when specific TopologyAttrs are present. The bare AtomGroup does not contain masses so many of the most used methods are invisible to the user (and a lot of our docs is broken, e.g., in the tutorial).

### Currently version of MDAnalysis:
develop, but also published 0.17.0

### PR DESCRIPTION
**Pull request by jbarnoud**
Fixes #1845 

Methods and properties defined by topology attributes and transplanted in various core classes (groups and `Universe`) do not appear in the documentation. Several solutions have been proposed to solve this problem:

* inspecting the code when generating the doc to programmatically add these methods to the doc: #2418, #2176.
* move the transplanted method back in the core classes: #2424.

Here, instead, I systematically transplant a dummy method in the core class for each method that is to be transplanted. The dummy method mimics the name, signature, and documentation of the original method, but always raises a `NoDataError`. The dummy methods (stubs) are picked up by sphinx and appear in the documentation. Also, the transplant methods from topology attributes that are not present in a given universe do not raise a confusing `AttributeError` any more, instead, they raise a `NoDataError` that specify which attribute is missing.

Compared to #2418, this method does not produce the cool looking tables and does not automatically tell that an attribute is required in the doc. It also has the same issue with reference duplicates (see https://github.com/MDAnalysis/mdanalysis/pull/2176#issuecomment-457108518). However, it does not require any manual operation.

The problem with the references is about one citation only, though.

Compared to #2424, it keeps all the benefit of the transplant as described in #601.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [X] Issue raised/referenced?

### CODE REVIEW

**Comment by jbarnoud** [comment]
I will move the transplant of the stub in the metaclass to keep all the logic together.

**Comment by jbarnoud** [comment]
This is what it looks like:

mda-doc

**Comment by jbarnoud** [comment]
I think this is ready for review. Maybe @lilyminium would have opinions about it.

**Review by lilyminium — APPROVED** [review]
This looks great, thank you @jbarnoud!  However, if it's not too much work to bring the Python 2 `funcsigs` back in, it would be great to backport this PR to master so the transplanted methods are documented in the 1.0.x docs too -- the methods do exist in that version as well.

File: package/MDAnalysis/core/topologyattrs.py [hunk 1]
```diff
@@ -163,6 +164,101 @@ def _wronglevel_error(attr, group):
...
 
 
+def _build_stub(method_name, method, attribute_name):
+    """
+    Build a stub for a transplanted method.
+
+    A transplanted stub is a dummy method that get attached to a core class
+    (usually from :mod:`MDAnalysis.core.groups`) and raise a
+    :exc:`NoDataError`.
+    The stub mimics the original method for everything that as traits with the
```
[the remark this answers is not in the document]
Comment (jbarnoud) [reply package/MDAnalysis/core/topologyattrs.py:174]:
```suggestion
    The stub mimics the original method for everything that has traits with the
```

**Comment by jbarnoud** [comment]
@lilyminium I added back funcsign and got rid of an f-string to help with backport. The patch should work on python 2.7, but I did not try as the rest of develop doesn't. The uncovered code is the import of funcsign as it would only run on python 2.

**Comment by lilyminium** [comment]
Thank you @jbarnoud for this nice solution!

**Comment by orbeckst** [comment]
Thank you @jbarnoud – makes me happy that this is working.

We can remove the funcsig stuff again – but it makes it easier to backport because we can cherry pick.

**Pull request merged**